cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/engine/sql/SQLEngineInput.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.api.engine.sql;
+
+import io.cdap.cdap.api.data.batch.Input;
+
+import java.util.Map;
+
+/**
+ * Sources that want to employ optimization by an SQL Engine may push SQLEngine-specific inputs, which include a
+ * fallback input in case the SQL Engine Input cannot be processed.
+ */
+public class SQLEngineInput extends Input {
+  private final String stageName;
+  private final String sqlEngineClassName;
+  private final Map<String, String> arguments;
+  private final Input fallbackInput;
```

Comment:
Same as the other comment. Fallback input was only used to have a hadoop-compatible input in case the SQL input failed to execute Since we don't support this anymore, the fallback input should be removed.